services-bytebuf-codec/src/main/java/io/scalecube/services/transport/api/ServiceMessageCodec.java

```diff
@@ -97,34 +185,19 @@ public ServiceMessage decode(ByteBuf dataBuffer, ByteBuf headersBuffer)
   }
 
   /**
-   * Decode message.
+   * Get a DataCodec for a content type.
    *
-   * @param message the original message (with {@link ByteBuf} data)
-   * @param dataType the type of the data.
-   * @return a new Service message that upon {@link ServiceMessage#data()} returns the actual data
-   *     (of type data type)
-   * @throws MessageCodecException when decode fails
+   * @param contentType the content type.
+   * @return a DataCodec for the content type or IllegalArgumentException is thrown if non exist
    */
-  public static ServiceMessage decodeData(ServiceMessage message, Type dataType)
-      throws MessageCodecException {
-    if (dataType == null
-        || !message.hasData(ByteBuf.class)
-        || ((ByteBuf) message.data()).readableBytes() == 0) {
-      return message;
+  private DataCodec getDataCodecByContentType(String contentType) {
```

Comment:
Maybe just `getDataCodec`. Parameter name `contentType` already saying that u want obtain something by content-type.

Reply:
Maybe rename `dataFormarOrDefault` to `contentTypeOfDefault`?

Reply:
Better not because everyone is referring to ServiceMessage.dataFormat .. as dataFormat.